File: VB.Net/DatingConsole.vb
Namespace DatingConsole
    
    Public Class Dating
        
        Public Shared Sub Main(ByVal args() As String)
            Dim Me As Person = New Person("Michael")
            Dim You As Person = New Person
            For Each Girl In You.GirlFriends
                If ((Me.GirlFriend Is Nothing)  _
                            AndAlso Me.CrushOn(Girl)) Then
                    Try 
                        If (Me.AskGirlOut(Girl) = "Accepted") Then
                            Me.Happy = true
                            Me.EnjoyLife
                        Else
                            Me.ReTry
                        End If
                        
                    Catch rx As RelationshipExistException
                        Throw rx.ReTry
                    End Try
                    
                End If
                
            Next
        End Sub
    End Class
    
    'Implement yours
    Public Class Person
        
        Private v As String
        
        Public Property GirlFriend As Object
            Get
            End Get
            Set
            End Set
        End Property
        
        Public Property GirlFriends As IEnumerable(Of Object)
            Get
            End Get
            Set
            End Set
        End Property
        
        Public Property Happy As Boolean
            Get
            End Get
            Set
            End Set
        End Property
        
        'Parameterless Constuct
        Public Sub New()
            MyBase.New
            
        End Sub
        
        Public Sub New(ByVal v As String)
            MyBase.New
            Me.v = Me.v
        End Sub
        
        Friend Function AskGirlOut(ByVal girl As Object) As String
            Throw New NotImplementedException
        End Function
        
        Friend Function CrushOn(ByVal girl As Object) As Boolean
            Throw New NotImplementedException
        End Function
        
        Friend Sub EnjoyLife()
            Throw New NotImplementedException
        End Sub
        
        Friend Sub PrepareForMarriage()
            Throw New NotImplementedException
        End Sub
        
        Friend Sub ReTry()
            Throw New NotImplementedException
        End Sub
    End Class
    
    'Implement your exception
    <Serializable()>  _
    Class RelationshipExistException
        Inherits Exception
        
        Public Sub New()
            MyBase.New
            
        End Sub
        
        Public Sub New(ByVal message As String)
            MyBase.New(message)
            
        End Sub
        
        Public Sub New(ByVal message As String, ByVal innerException As Exception)
            MyBase.New(message, innerException)
            
        End Sub
        
        Protected Sub New(ByVal info As SerializationInfo, ByVal context As StreamingContext)
            MyBase.New(info, context)
            
        End Sub
        
        Friend Function ReTry() As Exception
            Throw New NotImplementedException
        End Function
    End Class
End Namespace

File: c#/DatingConsole.cs
using System;
using System.Collections.Generic;
using System.IO;
using DatingImplementation;


namespace DatingConsole
{
    public class Dating
    {

        /* Dating Algorithm */
        public static void Main(string[] args)
        {
            Person Me = new Person("Michael");
            Person You = new Person();

            /* Loop through all the girls in the church */
            foreach (var Girl in You.GirlFriends)
            {
                /*if i dont have a girlfriend but am currently crushing on any. */
                if (Me.GirlFriend == null && Me.CrushOn(Girl))
                {
                    /* try and ask the girl out */
                    try
                    {
                        /* If she accepts my proposal */
                        if (Me.AskGirlOut(Girl) == "Accepted")
                        {
                            Me.Happy = true;
                            Me.EnjoyLife();
                        }
                        else
                        {
                            /* try a different girl. */
                            Me.ReTry();
                        }
                    }
                    catch (RelationshipExistException rx)
                    {
                        /* catch your dating mistakes and try again */
                        throw rx.ReTry();
                    }
                }
            }
        }




        public class Person
        {
            private string v;

            public Person()
            {

            }

            public Person(string v)
            {
                this.v = v;
            }

            public object GirlFriend { get; internal set; }

            public IEnumerable<object> GirlFriends { get; internal set; }

            public bool Happy { get; internal set; }

            internal string AskGirlOut(object girl)
            {
                throw new NotImplementedException();
            }

            internal bool CrushOn(object girl)
            {
                throw new NotImplementedException();
            }

            internal void EnjoyLife()
            {
                throw new NotImplementedException();
            }

            internal void PrepareForMarriage()
            {
                throw new NotImplementedException();
            }

            internal void ReTry()
            {
                throw new NotImplementedException();
            }
        }
    }



    ////Implement yours
    //namespace DatingImplementation
    //{
    //    public class Person
    //    {
    //        private string v;

    //        public Person()
    //        {

    //        }

    //        public Person(string v)
    //        {
    //            this.v = v;
    //        }

    //        public object GirlFriend { get; internal set; }

    //        public IEnumerable<object> GirlFriends { get; internal set; }

    //        public bool Happy { get; internal set; }

    //        internal string AskGirlOut(object girl)
    //        {
    //            throw new NotImplementedException();
    //        }

    //        internal bool CrushOn(object girl)
    //        {
    //            throw new NotImplementedException();
    //        }

    //        internal void EnjoyLife()
    //        {
    //            throw new NotImplementedException();
    //        }

    //        internal void PrepareForMarriage()
    //        {
    //            throw new NotImplementedException();
    //        }

    //        internal void ReTry()
    //        {
    //            throw new NotImplementedException();
    //        }
    //    }
    //}


    //Implement your exception
    [Serializable]
    internal class RelationshipExistException : Exception
    {
        public RelationshipExistException()
        {
        }

        public RelationshipExistException(string message) : base(message)
        {
        }

        public RelationshipExistException(string message, Exception innerException) : base(message, innerException)
        {
        }

        protected RelationshipExistException(SerializationInfo info, StreamingContext context) : base(info, context)
        {
        }

        internal Exception ReTry()
        {
            throw new NotImplementedException();
        }
    }
}